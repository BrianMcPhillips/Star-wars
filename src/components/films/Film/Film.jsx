import React from 'react';
import PropTypes from 'prop-types';

const Film = ({ title, url }) => {

  return (
    <>
      <h2>{title}</h2>
    </>
  );
};

Film.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Film;
