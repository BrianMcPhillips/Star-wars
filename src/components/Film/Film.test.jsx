import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Film from './Film';


describe('Film component', () => {
  afterEach(() => cleanup());
  it('renders Film', () => {
    const { asFragment } = render(
      <Film 
        name="A New Hope"
        url="https://swapi.dev/api/films/1/"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
