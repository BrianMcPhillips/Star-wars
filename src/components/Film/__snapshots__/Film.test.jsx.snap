// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film component renders Film 1`] = `
<DocumentFragment>
  <h2>
    A New Hope
  </h2>
</DocumentFragment>
`;
